app/ingestion/jd_mapping.json

```diff
@@ -0,0 +1,43 @@
+{
+  "mappings": {
+    "job_description": {
```

Comment:
there are basically - 3 things you can do with every field in ES -
1. Free form text search, in which case, the type should be ```text```
2. Aggregations only and not free form search, in which case, the type should be ```keyword```
3. Both aggregations and free form search, in which case, ES 5.x onwards, you can do this (free form search on foo and aggregations on foo.keyword)
```{
  "foo": {
    "type" "text",
    "fields": {
      "keyword": {
        "type": "keyword",
        "ignore_above": 256
      }
    }
  }
}```

Now, for every field in this mappings file, I am wondering what we want to be able to do - aggregations or free form search. Since it is better to keep the possibilities open now, we should map it like below for all fields.. thoughts?
```
{
  "foo": {
    "type" "text",
    "fields": {
      "keyword": {
        "type": "keyword",
        "ignore_above": 256
      }
    }
  }
}
```

Reply:
how is that materially different from what I've already done?  I already have `keyword` and `text` types both defined for the main text fields.

Are you just saying I should do that for _all_ the fields? I definitely would not want `keyword` type for the `summary` field in job descriptions. Up to you if you think we'd ever need it for any other fields I didn't specify this for

Reply:
for example - the field ```companyName``` -> if a company name has 2 tokens (Hewlett Packard), with the mapping you have, the inverted index for companyName would have ```Hewlett Packard``` as one word. But, with type as ```text```, the tokens are lower-cased, split and stored in inverted index - as ```hewlett``` and ```packard``` => when we do a free form search for only ```Hewlett``` or ```packard```, ES will find the document. So, I am saying may be better to stick to the same mapping for all the fields to support both free form search and aggregations

Reply:
I don't want to use the same mapping for all fields. `keyword` on `summary` is not worth the space.

But I'm open to using it for others, that's fine

Reply:
..and you are right - for the ```summary``` field, it does not make sense to use ```keyword``` - sorry, did not read that part earlier.